torchao/quantization/utils.py

```diff
@@ -349,6 +350,8 @@ def groupwise_affine_quantize_tensor_from_qparams(
     quant_max = 2 ** n_bit - 1
 
     int_data = quantize_affine(w, block_size, scales, zeros, output_dtype, quant_min, quant_max, zero_point_domain = ZeroPointDomain.FLOAT)
+    if TORCH_VERSION_AFTER_2_5:
+        int_data = (int_data[::, ::2] << 4 | int_data[::, 1::2]).to(torch.uint8)
```

Comment:
This should break on MPS backend, since __lshift__.Scalar is not currently implemented for MPS

Reply:
Is `int_data` in MPS device in this function? If so, we can make `int_data` in cpu device, then convert back to MPS device.

Reply:
@malfet landed https://github.com/pytorch/pytorch/pull/131813, so this won't be a problem anymore

Reply:
In any case, I learned from @malfet today (see his suggestion on line 203) that if instead of using << in here, we use torch.bitwise_left_shift(x, 4), it would be falling back to cpu even prior to his PR having landed.

Reply:
Thanks for the clarification. With https://github.com/pytorch/pytorch/pull/131813, `__lshift__.Scalar` has MPS dispatch now.